flower/utils/pg_storage.py

```diff
@@ -18,16 +22,40 @@
     (
         id SERIAL PRIMARY KEY,
         time TIMESTAMP NOT NULL,
-        data JSONB NOT NULL,
+        uuid VARCHAR(36) NOT NULL,
+        data TEXT NOT NULL,
```

Comment:
@johnarnold Dropping the GIN index and the JSONB is not an option for me, as I already depend on those for some data mining that is separate from Flower -- and I expect there will more of that in the future.

Reply:
@bosonogi Would you need the index if the table schema was expanded?  Are there specific columns you need?

JSONB is probably fine (it's about a 20-30% perf hit and a 200% increase in table size) but the GIN index calculating is pretty brutal as you get to large event sizes.  I can profile it some more for specifics after I figure out the write queueing.

Reply:
@johnarnold Well, the idea is to be able to access any fields from events in queries. While we don't need access to all of them right now, we do not know which of them we might need in the future.

Inserting in a separate thread should help with the increased computation cost. I've made a pull request to your branch with some quick changes to possibly help things move along -- and to be a bit more involved ;)  I hope I will have time to do more than just suggest untested code in the future.